fl4health/parameter_exchange/layer_exchanger.py

```diff
@@ -92,88 +91,76 @@ def pull_parameters(self, parameters: NDArrays, model: nn.Module, config: Option
         model.load_state_dict(current_state, strict=True)
 
 
-class NormDriftParameterExchanger(ParameterExchangerWithPacking[List[str]]):
+class DynamicLayerExchanger(PartialParameterExchanger[List[str]]):
     def __init__(
         self,
+        layer_selection_function: Callable[..., Tuple[NDArrays, List[str]]],
         norm_threshold: float,
         exchange_percentage: float = 0.1,
         normalize: bool = True,
         filter_by_percentage: bool = True,
+        select_drift_more: bool = True,
     ) -> None:
         """
         This exchanger selects those parameters that at the end of each training round, drift away (in l2 norm)
         from their initial values (at the beginning of the same round) by more than self.threshold.
         """
+        self.set_layer_selection_function(layer_selection_function)
         self.parameter_packer = ParameterPackerWithLayerNames()
         assert 0 < exchange_percentage <= 1
         assert 0 < norm_threshold
-        self.threshold = norm_threshold
+        self.norm_threshold = norm_threshold
         self.exchange_percentage = exchange_percentage
         self.set_normalization_mode(normalize)
         self.set_filter_mode(filter_by_percentage)
+        self.set_select_drift_more(select_drift_more)
+
+    def set_layer_selection_function(
+        self, layer_selection_function: Callable[..., Tuple[NDArrays, List[str]]]
```

Comment:
Correct me if I'm wrong, but below, this function actually has a fixed signature. That is, it takes 
`model, initial_model, self.exchange_percentage, self.normalize, self.select_drift_more`
So the type is `Callable[nn.Module, nn.Module, float, bool, bool, Tuple[NDArrays, List[str]]]`. It's a bit verbose, but you could create a type alias to simplify the annotation if you want.